src/bun.js/bindings/exports.zig

```diff
@@ -2601,93 +2785,193 @@ pub const ZigConsoleClient = struct {
                             .Float64Array => this.writeTypedArray(
                                 *@TypeOf(writer),
                                 &writer,
+                                Writer,
+                                writer_,
                                 f64,
                                 @alignCast(std.meta.alignment([]f64), std.mem.bytesAsSlice(f64, slice)),
                                 enable_ansi_colors,
                             ),
                             .BigInt64Array => this.writeTypedArray(
                                 *@TypeOf(writer),
                                 &writer,
+                                Writer,
+                                writer_,
                                 i64,
                                 @alignCast(std.meta.alignment([]i64), std.mem.bytesAsSlice(i64, slice)),
                                 enable_ansi_colors,
                             ),
                             .BigUint64Array => this.writeTypedArray(
                                 *@TypeOf(writer),
                                 &writer,
+                                Writer,
+                                writer_,
                                 u64,
                                 @alignCast(std.meta.alignment([]u64), std.mem.bytesAsSlice(u64, slice)),
                                 enable_ansi_colors,
                             ),
 
                             // Uint8Array, Uint8ClampedArray, DataView, ArrayBuffer
-                            else => this.writeTypedArray(*@TypeOf(writer), &writer, u8, slice, enable_ansi_colors),
+                            else => this.writeTypedArray(*@TypeOf(writer), &writer, Writer, writer_, u8, slice, enable_ansi_colors),
                         }
                     }
 
-                    writer.writeAll(" ]");
+                    if (this.snapshot_format) {
+                        if (slice.len > 0) {
+                            writer.writeAll("\n");
+                            this.writeIndent(Writer, writer_) catch {};
+                            writer.writeAll("]");
+                            if (this.indent == 0) {
+                                writer.writeAll("\n");
+                            }
+                        } else {
+                            writer.writeAll("]");
+                        }
+                    } else {
+                        writer.writeAll(" ]");
+                    }
                 },
                 else => {},
             }
         }
 
-        fn writeTypedArray(this: *ZigConsoleClient.Formatter, comptime Writer: type, writer: Writer, comptime Number: type, slice: []const Number, comptime enable_ansi_colors: bool) void {
+        fn writeTypedArray(this: *ZigConsoleClient.Formatter, comptime WriterWrapped: type, writer: WriterWrapped, comptime Writer: type, writer_: Writer, comptime Number: type, slice: []const Number, comptime enable_ansi_colors: bool) void {
```

Comment:
This is going to generate like 50 copies of this function

can the snapshot version be hoisted up